jupyter_server/extension/application.py

```diff
@@ -419,6 +419,8 @@ def start(self):
     def stop(self):
         """Stop the underlying Jupyter server.
         """
+        if hasattr(self, 'stop_extension'):
```

Comment:
I think we should add an async stop_extension() method to ExtensionApp that passes by default, rather than checking if it exists.